docs/pytorch_profiler.md

```diff
@@ -2,124 +2,105 @@
+
+### Duration based vs iteration based traces
+By default the traces are captured for a specific time duration. One can configure this trace duration in ms using the `--duration-ms` flag of the dyno command line tool.
+
+A common requirement is to capture 3-5 trainging iterations of your AI model. This is now possible as PyTorch includes a profiling hook in the optimizer. Generally speaking, training involves at least one optimzer operation in each iteration.
 
-### Triggering traces on a remote node
-To trigger trace on a remote node you can use the `--host-name` flag.
+We can make a request to capture iteration based traces using the `--iterations` flag.
 ```bash
-dyno --host-name a100-st-p4d24xlarge-42 gputrace --log-file /path/to/log/file/libkineto_trace.json
+$> dyno gputrace --log-file /tmp/libkineto_trace.json --iterations 5
 ```
+For a working example of the above you can run the `scripts/pytorch/xor.py` that contains a training loop with an optimizer. If PyTorch profiler is unable to capture an iteration based trace it **defaults back to a duration based trace**.
 
-It is likely that a single node can run multiple jobs. Pytorch/Kineto can recognize job IDs
-of processes run under the SLURM job scheduler. You can pass a job id to filter requests as follows.
+### On-demand tracing on a remote node
+To trigger a PyTorch trace on a remote node you can use the `--host-name` flag. Note that dynolog should be running on that remote node.
 ```bash
-dyno --host-name a100-st-p4d24xlarge-42 gputrace --job-id 4242 --log-file /path/to/log/file/libkineto_trace.json
+dyno --host-name remote_host gputrace --log-file /path/to/log/file/libkineto_trace.json
 ```
+**Tip**: If your cluster supports NFS you can set the log file path to a directory backed by NFS.
 
-## Tracing a distributed application
-Lastly, this section talks about how to profile a full scale traing jobs containing multiple nodes.
-We provide an example with SLURM, a job scheduler commonly used in HPC environments.
-
-### Launching job with dynolog
-In case dynolog is not deployed on your cloud/cluster you can launch the dynolog binary
-with your job. The script `run_with_dyno_wrapper.sh` can be used for this purpose.
-
-This script will spin up a dynolog process first and then run your job's command. It will also
-setup the required environment variables. An example is shown below:
+Also, it is possible that a single node can run multiple cluster jobs. Pytorch/Kineto can recognize job IDs
+of processes run under the SLURM job scheduler. One can pass a job id to filter processes as follows.
 ```bash
-srun ./scripts/slurm/run_with_dyno_wrapper.sh <desired launch cmd>
-# example
-srun ./scripts/slurm/run_with_dyno_wrapper.sh python3 ./scripts/pytorch/linear_model_example.p
-
+dyno --host-name remote_host gputrace --job-id 4242 --log-file /path/to/log/file/libkineto_trace.json
 ```
 
+## Co-ordinated Tracing on a distributed application
```

Comment:
nit: s/co-ordinated/coordinated/

Reply:
TIL https://onlinewritingtraining.com.au/coordinate-co-ordinate/ the hypen is being dropped